### PR TITLE
Allow query matcher to use repeated query string parameters

### PR DESCRIPTION
**Pull request by lodrantl**
Pretty self explanatory PR.

```
fetchMock.mock({
  query: {
    a: ["b", "c"]
  }
}....
```
which will match:
http://my.domain.url/api?a=b&a=c

But not
http://my.domain.url/api?a=b
http://my.domain.url/api?a=b&a=c&a=d

### CODE REVIEW

**Comment by wheresrhys** [comment]
Interesting. Question - what should be the behaviour for `http://my.domain.url/api?a[]=b&a[]=c&a[]=d` ?

**Comment by lodrantl** [comment]
Are you asking if  `http://my.domain.url/api?a[]=b&a[]=c&a[]=d`
would be accepted for matcher
```
fetchMock.mock({
  query: {
    a: ["b", "c", "d"]
  }
}....
```
Right now it is not, I think PHP would accept it right?
As far as a mocking framework goes you could always use
```
fetchMock.mock({
  query: {
    "a[]": ["b", "c", "d"]
  }
}....
```

**Comment by wheresrhys** [comment]
Yeah quite right - just been reading up on the lack of any spec on multi-value query strings, so keeping the behaviour simple and just treating all keys as strings is right.

File: docs/_api-mocking/mock_matcher.md [hunk 1]
```diff
@@ -150,6 +150,8 @@ parameters:
         examples:
           - |-
             {"q": "cute+kittenz", "format": "gif"}
+          - |-
+            {"tags": ["cute", "kittenz"]}
```
Comment (wheresrhys) [inline docs/_api-mocking/mock_matcher.md:154]: Could you update the docs just above to say: 
> Match only requests that have these query parameters set (in any order). In order to match keys which are used multiple times use an array of strings.

File: src/lib/matchers.js [hunk 2]
```diff
@@ -85,7 +85,17 @@ const getQueryStringMatcher = ({ query: expectedQuery }) => {
 		const query = querystring.parse(getQuery(url));
 		debug('  Expected query parameters:', expectedQuery);
 		debug('  Actual query parameters:', query);
-		return keys.every((key) => query[key] === expectedQuery[key]);
+		return keys.every((key) => {
+			const value = query[key];
+			const expectedValue = expectedQuery[key];
+			if (Array.isArray(value) && Array.isArray(expectedValue)) {
```
Comment (wheresrhys) [inline src/lib/matchers.js:91]: What about if my mock is `key: ['val']`. I _think_ this won't match `?key=val` as the parsed query string will be `{key: 'val'}`

So needs a test for that scenario. And I think the clearest way to write the matcher is to coerce both value and expectedValue to arrays up front, then compare the arrays

Comment (lodrantl) [reply src/lib/matchers.js]: True, I think it should be okay now. Looks a bit cleaner too.

**Comment by ayZagen** [comment]
I believe #555 covers this also.

**Comment by wheresrhys** [comment]
@ayZagen I think I will merge this PR as the tests cover an interesting edge case, and I think the object comparison is nicer - hopefully it is not too much work to rebase onto it

@lodrantl  - thanks for the changes. Will release as v9.7.0

**Comment by ayZagen** [comment]
@wheresrhys This would fail with query params containing `null` and `boolean` values
